### PR TITLE
Ui tag colour

### PR DESCRIPTION
**Pull request by ChangDarren**
Add UI Tag Colour feature by creating an additional attribute in the Tag. Feature can be used in add and edit command. If the colour is not specified, the hash of the tag name will be used to determine a random colour.

Note: User guide is not updated as the AB4 add and edit feature is different from TopDeck's feature

### CODE REVIEW

File: src/main/java/seedu/address/model/tag/Tag.java [hunk 1]
```diff
@@ -10,26 +10,55 @@
...
+        }
+        return matchesPattern && isValidTagColor(attributes[1]);
+    }
+
+    /**
+     * Returns true if the given tag color is a valid color.
+     */
+    private static boolean isValidTagColor(String test) {
+        for (String color: TAG_COLOR_STYLES) {
+            if (test.compareTo(color) == 0) {
```
Comment (xsot) [inline src/main/java/seedu/address/model/tag/Tag.java]: Use `equals` to test for equality.

**Review by xsot — COMMENTED** [review]
Is it possible to store multiple attributes in a tag? Storing a tag's name together with its color seems clumsy. This also doesn't scale well if we decide to add more optional properties in the future.

File: src/main/java/seedu/address/model/tag/Tag.java [hunk 2]
```diff
@@ -26,7 +26,7 @@ public Tag(String tagName) {
     }
 
     /**
-     * Returns true if a given string is a valid tag name.
+     * Returns true if a given string is a valid tag name and color.
```
Comment (xsot) [inline src/main/java/seedu/address/model/tag/Tag.java]: Outdated comment.

File: src/main/java/seedu/address/model/tag/Tag.java [hunk 3]
```diff
@@ -48,7 +48,7 @@ public int hashCode() {
      * Format state as text for viewing.
      */
     public String toString() {
-        return '[' + tagName + ']';
+        return tagName;
```
Comment (xsot) [inline src/main/java/seedu/address/model/tag/Tag.java]: Not sure why you changed this.

**Review by xsot — CHANGES_REQUESTED** [review]
Some minor comments. Also, your PR is failing checkstyle.

**Comment by ChangDarren** [comment]
Edited the check style so that the PR works. Decided to localise the tag colour as too many things will be changed if I were to create a new JSON for tag mapping.

**Pull request merged**